docs/tutorials/semantic_search/asymmetric_embedding_model.md

```diff
@@ -0,0 +1,541 @@
+# Tutorial: Running Asymmetric Semnantic Search within OpenSearch
+
+This tutorial demonstrates how generating text embeddings using an asymmetric embedding model in OpenSearch. The embeddings will be used
+to run semantic search, implemented using a Docker container. The example model used in this tutorial is the multilingual
+`intfloat/multilingual-e5-small` model from Hugging Face. 
+You will learn how to prepare the model, register it in OpenSearch, and run inference to generate embeddings.
+
+> **Note**: Make sure to replace all placeholders (e.g., `your_`) with your specific values.
+
+---
+
+## Prerequisites
+
+- Docker Desktop installed and running on your local machine.
+- Basic familiarity with Docker and OpenSearch.
+- Access to the Hugging Face `intfloat/multilingual-e5-small` model (or another model of your choice).
+---
+
+## Step 1: Spin up a Docker OpenSearch cluster
+
+To run OpenSearch in a local development environment, you can use Docker and a preconfigured `docker-compose` file.
+
+### a. Create a Docker Compose File
+
+You can use this sample [file](https://opensearch.org/docs/latest/install-and-configure/install-opensearch/docker/#sample-docker-compose-file-for-development) as an example.
+Once your `docker-compose.yml` file is created, run the following command to start OpenSearch in the background:
+
+```
+docker-compose up -d
+```
+
+
+### b. Update cluster settings
+
+Ensure your cluster is configured to allow registering models. You can do this by updating the cluster settings using the following request:
+
+```
+PUT _cluster/settings
+{
+  "persistent": {
+    "plugins.ml_commons.allow_registering_model_via_url": "true",
+    "plugins.ml_commons.only_run_on_ml_node": "false",
+    "plugins.ml_commons.model_access_control_enabled": "true",
+    "plugins.ml_commons.native_memory_threshold": "99"
+  }
+}
+```
+
+This configuration ensures that OpenSearch can accept machine learning models from external URLs and can run models across non-ML nodes.
+
+
+---
+
+## Step 2: Prepare the model for OpenSearch
+
+In this tutorial, you’ll use the Hugging Face `intfloat/multilingual-e5-small` model, which is capable of generating multilingual embeddings. Follow these steps to prepare and zip the model for use in OpenSearch.
+
+### a. Clone the model from Hugging Face
+
+To download the model, use the following steps:
+
+1. Install Git Large File Storage (LFS) if you haven’t already:
```

Comment:
Seems like we should release this model from our [pre-trained model repository](https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/) so that customer can easily register this model?

Reply:
What is the process to introduce the model (to the model repository)?

Reply:
We can discuss offline more about this. But this is an example of the [model tracing workflow](https://github.com/opensearch-project/opensearch-py-ml/actions/runs/10357363209) from opensearch-py-ml